src/pages/Bridge/index.tsx

```diff
@@ -81,48 +120,70 @@ const BridgePage: React.FC = () => {
     accountAddress,
   });
 
+  const { data: getXvsBridgeFeeEstimationData } = useGetXvsBridgeFeeEstimation(
+    {
+      accountAddress: accountAddress || '',
+      destinationChain: toChainId!,
+      amountMantissa,
+    },
+    {
+      enabled: !!accountAddress && amountMantissa.gt(0),
+    },
+  );
+
+  const [bridgeEstimatedFeeMantissa, brigeEstimatedFeeTokens] = useMemo(() => {
+    const feeMantissa = getXvsBridgeFeeEstimationData?.estimationFeeMantissa || new BigNumber(0);
+    return [
+      feeMantissa,
+      convertMantissaToTokens({
+        value: feeMantissa,
+        token: nativeToken,
+      }),
+    ];
+  }, [getXvsBridgeFeeEstimationData, nativeToken]);
+
   const handleChainFieldChange = useCallback(
-    ({ fromChainId, toChainId }: { fromChainId?: ChainId; toChainId?: ChainId }) => {
+    ({ newFromChainId, newToChainId }: { newFromChainId?: ChainId; newToChainId?: ChainId }) => {
```

Comment:
I wouldn't change the names of the properties, as it can reduce reusability (when passing arguments to the function for example).
To prevent conflict with higher-level variables/props, you can instead use the object input directly:
```suggestion
    (input: { fromChainId?: ChainId; toChainId?: ChainId }) => {
```

Reply:
I actually kept this, not so much because of the name conflict, but because of the callbacks:

```
callback: () => {
  setValue('fromChainId', newFromChainId); -> changing this to input.fromChainId will throw
},
```
Type narrowing does not help inside the callbacks, so you end up having to add even more code to make sure it knows the key inside `input` has a value. Destructuring `input` would work as expected, but then you have clashing names.

Reply:
I see, that makes sense. Having to assert types would reduce code readability.